strapi/strapi
# Warn if trying to use Strapi v3's `enabled` option to disable the CORS middleware

### What does it do?

Warn if trying to use Strapi v3's `enabled` option to disable the CORS middleware in v4

### Why is it needed?

At my work, we used the `strapi::cors` enabled option with an envvar to allow any origin in development, but disallow it in production:

```ts
// config/middlewares.js
module.exports = [
  ...,
    {
    name: 'strapi::cors',
    config: {
      enabled: process.env.STRAPI_DANGEROUSLY_ALLOW_ANY_ORIGIN === 'true',
      headers: '*',
      origin: [
        '*'
      ]
    }
  },

  ...
```

This worked fine in v3. However, when moving to v4 this *silently* caused insecure CORS headers in production because v4 doesn't support the `enabled` flag.

Granted, copying snippets from message boards is not as noble an approach as reading Strapi's official docs, but those come up first if you search the internet for `strapi cors`.

### How to test it?

Replace `'strapi::cors'` in `examples/getstarted/config/middlewares.js` with this snippet:
```ts
  {
    name: 'strapi::cors',
    config: {
      enabled: false, // does nothing except raise a warning
      headers: '*',
      origin: [
        '*'
      ]
    }
  },
```
Then, run:
```
yarn setup
cd examples/getstarted
yarn develop
```
This should show the warning.
